Update specs with batching and lack of vote extensions (#393)
anoma/namada: documentation/specs/src/interoperability/ethereum-bridge/transfers_to_namada.md

# Transferring assets from Ethereum to Namada

In order to facilitate transferring assets from Ethereum to Namada, There 
 will be two internal accounts with associated native validity predicates:

- `#EthBridge` - Controls the `/eth_msgs/` [storage](ethereum_events_attestation.md/#storage) 
- and ledgers of balances
  for wrapped Ethereum assets (ERC20 tokens) structured in a
  ["multitoken"](https://github.com/anoma/anoma/issues/1102) hierarchy
- `#EthBridgeEscrow` which will hold in escrow wrapped Namada tokens which have
  been sent to Ethereum.

#### Wrapped ERC20

If an ERC20 token is transferred to Namada, once the associated 

> `TransferToNamada` Ethereum event is included into Namada, validators mint 
> the  appropriate amount to the corresponding  multitoken balance key for 
> the receiver.

Reviewer: ```suggestion
`TransferToNamada` Ethereum event is included into Namada, validators mint 
the appropriate amount to the corresponding  multitoken balance key for 
the receiver, or release the escrowed native Namada token.
```

```rust
pub struct EthAddress(pub [u8; 20]);

> /// Represents Ethereum assets on the Ethereum blockchain
> pub enum EthereumAsset {
>     /// An ERC20 token and the address of its contract
>     ERC20(EthAddress),
> }
> 
> /// An event transferring some kind of value from Ethereum to Anoma
> pub struct TransferToNamada {
>     /// Quantity of ether in the transfer
>     pub amount: Amount,
>     /// Address on Ethereum of the asset
>     pub asset: EthereumAsset,
>     /// The Namada address receiving wrapped assets on Anoma
>     pub receiver: Address,
> }

Reviewer: ```suggestion
/// An event transferring some kind of value from Anoma to Ethereum
pub struct TransferToEthereum {
    /// Quantity of wrapped Asset in the transfer
    pub amount: Amount,
    /// Address of the smart contract issuing the token
    pub asset: EthAddress,
    /// The address receiving assets on Ethereum
    pub receiver: EthAddress,
}
```
to update: `asset` will always be an ERC20 address (at least for this initial version of the bridge). In the case of transferring NAM to Ethereum, `asset` is the ERC20 address of wNAM which is set when we deploy the Ethereum bridge smart contracts.

https://github.com/anoma/ethereum-bridge/blob/a1fcd58f5f51f161d9d70a142aa0e9bc0ae5a52e/scripts/deploy.js#L139-L140

Author: This should be a transfer to Namada, not Ethereum.

Reviewer: sorry the struct should be
```rust
pub struct TransferToNamada {
    /// Quantity of the ERC20 token in the transfer
    pub amount: Amount,
    /// Address of the smart contract issuing the token
    pub asset: EthAddress,
    /// The address receiving wrapped assets on Anoma
    pub receiver: Address,
}
```

```

##### Example

For 10 DAI i.e. ERC20([0x6b175474e89094c44da98b954eedeac495271d0f](https://etherscan.io/token/0x6b175474e89094c44da98b954eedeac495271d0f)) to `atest1v4ehgw36xue5xvf5xvuyzvpjx5un2v3k8qeyvd3cxdqns32p89rrxd6xx9zngvpegccnzs699rdnnt`
```

> #EthBridge
>     /erc20
>         /0x6b175474e89094c44da98b954eedeac495271d0f
>             /balances
>                 /atest1v4ehgw36xue5xvf5xvuyzvpjx5un2v3k8qeyvd3cxdqns32p89rrxd6xx9zngvpegccnzs699rdnnt 

Reviewer: ```suggestion
#EthBridge
    /ERC20
        /0x6b175474e89094c44da98b954eedeac495271d0f
            /balance
                /atest1v4ehgw36xue5xvf5xvuyzvpjx5un2v3k8qeyvd3cxdqns32p89rrxd6xx9zngvpegccnzs699rdnnt 
```

                += 10
```

#### Namada tokens

Any wrapped Namada tokens being redeemed from Ethereum must have an 
equivalent amount of the native token held in escrow by `#EthBridgeEscrow`.
Once the associated`TransferToNamada` Ethereum event is included into 
Namada, validators should simply make a transfer from `#EthBridgeEscrow` to 

> the `receiver` for the appropriate amount and asset.

Reviewer: I can spec this out further but we could actually do the release of escrowed NAM as a free derived transaction. In which case the sender only has to make the transaction on the Ethereum side for redeeming wNAM and everything else will happen automatically (as is the case with sending other ERC20s).

Author: It is a free derived transaction. I'll make that wording clearer.
